### PR TITLE
AirLoopHVAC:UnitaryHeatCool:VAVChangeoverBypass with CoilSystem:Cooling:DX:HeatExchangerAssisted shows false fatal error

### LINKED ISSUE #10253: AirLoopHVAC:UnitaryHeatCool:VAVChangeoverBypass fails with CoilSystem:Cooling:DX:HeatExchangerAssisted
Issue overview
--------------
Recent refactor incorrectly used the DX coil nodes to represent the HX assisted object nodes resulting in a severe/fatal error.

    ** Severe  ** AirLoopHVAC:UnitaryHeatCool:VAVChangeoverBypass: AIR LOOP HVAC UNITARY HEAT COOL VAVCHANGEOVER BYPASS 3
    **   ~~~   ** Illegal cooling coil placement. The cooling coil inlet node name must be the same as the mixed air node specified in the Outdoor Air Mixer Name = AIR LOOP HVAC UNITARY HEAT COOL VAVCHANGEOVER BYPASS 3 OUTDOOR AIR MIXER when draw through Supply Air Fan Placement is selected.
    **  Fatal  ** GetCBVAV: Errors found in getting AirLoopHVAC:UnitaryHeatCool:VAVChangeoverBypass input.

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added ([in-CBVAV-CoilPlacement.idf.txt](https://github.com/NREL/EnergyPlus/files/12751859/in-CBVAV-CoilPlacement.idf.txt))
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
 - Fixes #10253
 - Incorrect node index was used for HXAssisted cooling coil for node name checks resulting in a fatal error

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/HVACUnitaryBypassVAV.cc [hunk 1]
```diff
@@ -739,8 +739,12 @@ namespace HVACUnitaryBypassVAV {
...
                             ErrorsFound = true;
                         } else {
-                            thisCBVAV.DXCoilInletNode = state.dataDXCoils->DXCoil(thisCBVAV.DXCoolCoilIndexNum).AirInNode;
-                            thisCBVAV.DXCoilOutletNode = state.dataDXCoils->DXCoil(thisCBVAV.DXCoolCoilIndexNum).AirOutNode;
+                            // the "coil" nodes are actually the parent nodes of the wrapped HX
+                            thisCBVAV.DXCoilInletNode =
+                                HVACHXAssistedCoolingCoil::GetCoilInletNode(state, thisCoolCoilType, thisCBVAV.DXCoolCoilName, DXErrorsFound);
+                            thisCBVAV.DXCoilOutletNode =
+                                HVACHXAssistedCoolingCoil::GetCoilOutletNode(state, thisCoolCoilType, thisCBVAV.DXCoolCoilName, DXErrorsFound);
+                            // the DX coil holds the condenser inlet node number
```
Comment (rraustad) [inline src/EnergyPlus/HVACUnitaryBypassVAV.cc:747]: This was just a dumb mistake during a refactor exercise. DXCoilInletNode here refers to the parent object node, not the coil inside of the parent.

Comment (Myoldmopar) [reply src/EnergyPlus/HVACUnitaryBypassVAV.cc:747]: :+1:

**Review by Myoldmopar — APPROVED** [review]

**Comment by Myoldmopar** [comment]
Windows is just dumb sometimes, it's unrelated.  Otherwise CI is totally happy, thanks @rraustad

**Comment by Myoldmopar** [comment]
Everything is happy with both this branch and the branch from #10251 pulled in.  I'm merging both of them in now.

**Pull request merged**